plugins/3MFReader/ThreeMFWorkspaceReader.py

```diff
@@ -1243,3 +1249,26 @@ def _getMaterialLabelFromSerialized(serialized):
         metadata = data.iterfind("./um:metadata/um:name/um:label", {"um": "http://www.ultimaker.com/material"})
         for entry in metadata:
             return entry.text
+
+    @staticmethod
+    def _parse_packages_metadata(archive: zipfile.ZipFile) -> List[Dict[str, str]]:
+        try:
+            package_metadata = json.loads(archive.open("Metadata/packages.json").read().decode("utf-8"))
+            return package_metadata["packages"]
+        except Exception:
+            Logger.error("Failed to load packes metadata from .3mf file")
+        return []
+
+
+    @staticmethod
+    def _filter_missing_package_metadata(package_metadata: List[Dict[str, str]]) -> List[Dict[str, str]]:
+        """Filters out installed packages from package_metadata"""
+        missing_packages = []
+        package_manager = cast(CuraPackageManager, CuraApplication.getInstance().getPackageManager())
+
+        for package in package_metadata:
+            package_id = package["id"]
+            if not package_manager.isPackageInstalled(package_id):
```

Comment:
What do we say, or does happen to (if anything, as it's a bit of an edge case and can just become a 'known issue') to a user that has just installed a (material) package, but declined to restart Cura, and then opened a file that has that material in it?

Reply:
That is indeed a good idea to inform the user in that case. Opened a popup when the user closes the install missing packaged dialog: https://github.com/Ultimaker/Cura/pull/12434/commits/76180eee011c4c51ec88c83913b399ba700fb268. I was a bit annoying to implement as i didn't want to copy the show warning function so I passed a reference to that function to the constructor of the InstallMissingPackages manager. It's not very elegant but it prevented some copy pasting.

Reply:
Just now I thought of another solution. Why not immediately show a warning if we detect missing materials; It is not very distracting as the pop doesn't show in the active window (the active window is the open file summary dialog). And it doesn't matter if it isn't removed, it essentially removed in the correct flow where the user clicks the "close cura" buttin in the install missing packages dialog.

Reply:
The new solution proposed sounds good as well. I'm going to continue with this review (and merge if all goes well), so we don't lose any time. You can do this as a small fry afterwards if you're up to it.